app/controllers/data_explorer/query_controller.rb

```diff
@@ -0,0 +1,236 @@
+class DataExplorer::QueryController < ::ApplicationController
+  requires_plugin DataExplorer.plugin_name
+
+  before_action :check_enabled
+  before_action :set_group, only: %i(group_reports_index group_reports_show group_reports_run)
+  before_action :set_query, only: %i(group_reports_show group_reports_run)
+
+  attr_reader :group, :query
+
+  def check_enabled
+    raise Discourse::NotFound unless SiteSetting.data_explorer_enabled?
+  end
+
+  def set_group
+    @group = Group.find_by(name: params["group_name"])
+  end
+
+  def set_query
+    @query = DataExplorer::Query.find(params[:id])
+  end
+
+  def index
+    queries = DataExplorer::Query.where(hidden: false).order(:last_run_at, :name).to_a
+
+    database_queries_ids = DataExplorer::Query.pluck(:id)
+    Queries.default.each do |params|
+      attributes = params.last
+      query = DataExplorer::Query.new
+      query.id = attributes["id"]
+      query.sql = attributes["sql"]
+      query.name = attributes["name"]
+      query.description = attributes["description"]
+      query.user_id = Discourse::SYSTEM_USER_ID.to_s
+      queries << query unless database_queries_ids.include?(query.id)
+    end
+
+    render_serialized queries, DataExplorer::QuerySerializer, root: 'queries'
+  end
+
+  skip_before_action :check_xhr, only: [:show]
```

Comment:
Generally I prefer to move this to the top of the controller so that we can see all the callbacks in a single place.